### PR TITLE
Add ContainsHeader method to check custom header presence

### LINKED ISSUE #22: Can not check the presence of an optional custom header
Use case: I need to get a custom header and this header is optional.

Currently there is no way to check the presence of a custom header so I resorted to wrap the indexed access (in HeaderCollection) in a try block and catch the exception thrown (if the header is missing from the internal dictionary) but it's not optimal.

### PR DESCRIPTION
**Pull request by lbragaglia**
Adds a ContainsHeader to HeaderCollection and also to container message classes, i. e. consumed and outbound message (resolves #22).

### CODE REVIEW

File: src/Carrot.Tests/OutboundEnveloping.cs [hunk 1]
```diff
@@ -303,6 +303,19 @@ public void MessageExpiration()
                                                           StubDateTimeProvider().Object,
                                                           new Mock<INewId>().Object);
             Assert.Equal("18000", properties.Expiration);
+        }
+
+        [Fact]
+        public void CustomHeader()
+        {
+            var message = new OutboundMessage<Bar>(new Bar());
+            const string key = "a";
```
Comment (naighes) [inline src/Carrot.Tests/OutboundEnveloping.cs]:
```suggestion
            const String key = "a";
```

File: src/Carrot.Tests/OutboundEnveloping.cs [hunk 2]
```diff
@@ -303,6 +303,19 @@ public void MessageExpiration()
...
                                                           new Mock<INewId>().Object);
             Assert.Equal("18000", properties.Expiration);
+        }
+
+        [Fact]
+        public void CustomHeader()
+        {
+            var message = new OutboundMessage<Bar>(new Bar());
+            const string key = "a";
+            const string value = "b";
```
Comment (naighes) [inline src/Carrot.Tests/OutboundEnveloping.cs]:
```suggestion
            const String value = "b";
```

File: src/Carrot/Messages/Message.cs [hunk 3]
```diff
@@ -9,6 +9,8 @@ public abstract class Message<TMessage> : IMessage
 
         public abstract HeaderCollection Headers { get; }
 
+        public bool ContainsHeader(String key) => Headers.ContainsHeader(key);
```
Comment (naighes) [inline src/Carrot/Messages/Message.cs]:
```suggestion
        public Boolean ContainsHeader(String key) => Headers.ContainsHeader(key);
```

**Review by naighes — CHANGES_REQUESTED** [review]
Looks fine!
We're used to keep close to [.NET built'in types](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/built-in-types-table) and aliasing is "discouraged".
Please, have a look to my suggestions.

Thanks!

**Comment by lbragaglia** [comment]
Please let me know if you prefer squashing into one commit before merging.

**Comment by naighes** [comment]
> Please let me know if you prefer squashing into one commit before merging.

Yeah, that'd be great!

**Pull request merged**